pkg/ingester/errors.go

```diff
@@ -135,3 +136,95 @@ func formatMaxMetadataPerMetricError(limits *validation.Overrides, labels labels
 	))
 	return makeMetricLimitError(labels, err)
 }
+
+type ingesterErrSamplers struct {
+	sampleTimestampTooOld             *log.Sampler
+	sampleTimestampTooOldOOOEnabled   *log.Sampler
+	sampleTimestampTooFarInFuture     *log.Sampler
+	sampleOutOfOrder                  *log.Sampler
+	sampleDuplicateTimestamp          *log.Sampler
+	maxSeriesPerMetricLimitExceeded   *log.Sampler
+	maxMetadataPerMetricLimitExceeded *log.Sampler
+	maxSeriesPerUserLimitExceeded     *log.Sampler
+	maxMetadataPerUserLimitExceeded   *log.Sampler
+}
+
+func newIngesterErrSamplers(freq int64) ingesterErrSamplers {
+	if freq == 0 {
+		return ingesterErrSamplers{}
+	}
```

Comment:
While valid, I'd remove this from here. `log.NewSamler(0)` will return `nil` with same effect, and it's better to have logic in a single place (sampler knows how to sample freq=0, there's no need to give this dumb struct responibility for that)